src/CentrifugeRouter.sol

```diff
@@ -2,97 +2,186 @@
 pragma solidity 0.8.21;
 
 import {Auth} from "src/Auth.sol";
-import {Multicall} from "src/Multicall.sol";
+import {MathLib} from "src/libraries/MathLib.sol";
 import {SafeTransferLib} from "src/libraries/SafeTransferLib.sol";
+import {IERC20, IERC20Permit, IERC20Wrapper} from "src/interfaces/IERC20.sol";
 import {IERC7540Vault} from "src/interfaces/IERC7540.sol";
-import {IERC20} from "src/interfaces/IERC20.sol";
 import {ICentrifugeRouter} from "src/interfaces/ICentrifugeRouter.sol";
 import {IPoolManager} from "src/interfaces/IPoolManager.sol";
 
-contract CentrifugeRouter is Auth, Multicall, ICentrifugeRouter {
+interface EscrowLike {
+    function approve(address token, address spender, uint256 value) external;
+}
+
+contract CentrifugeRouter is Auth, ICentrifugeRouter {
+    address public immutable escrow;
```

Comment:
Escrow is required because intended usage of the Router allows moving funds out of it (using `_approveMax` in e.g. `wrap`), so we have to separate the locked request funds to not expose them.